### DISCUSSION TITLE
Is there an example to check the registration status of an IoT edge device registered to an iot hub using device provisioning service in c?

### DISCUSSION THREAD

**Discussion by Arjun-Sivasree** [Azure/azure-iot-sdk-c#2453] [Q&A]
https://learn.microsoft.com/en-us/rest/api/iot-dps/device/runtime-registration/device-registration-status-lookup#deviceregistrationresult
As per above link, I understand that there is an api to check the device registration status in iot hub.

But is there an example which I can refer on how to use this api in c? I could not find anything in azure iot sdk c documentation or samples.

**Comment by danewalton-msft** [comment]
Hi @Arjun-Sivasree 
You can get this kind of capability by using the `PROV_DEVICE_CLIENT_REGISTER_STATUS_CALLBACK`. Possible values returned from this callback can be found here:

https://github.com/Azure/azure-iot-sdk-c/blob/b576b03d0d576ad4818cc27e6ab8996b2f92e78f/provisioning_client/inc/azure_prov_client/prov_device_ll_client.h#L41-L47

So while the status lookup isn't done synchronously by the user application, it is notified of the progress of the provisioning.

Let us know how else we can help here.
